core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -433,4 +440,46 @@ public final BuilderT defaultsMode(DefaultsMode defaultsMode) {
     public final void setDefaultsMode(DefaultsMode defaultsMode) {
         defaultsMode(defaultsMode);
     }
+
+    /**
+     * If the region is a FIPS pseudo region (contains "fips"), this method returns a pair of values, the left side being the
+     * region with the "fips" string removed, and the right being {@code true}. Otherwise, the region is returned
+     * unchanged, and the right will be empty.
+     */
+    private static Pair<Region, Optional<Boolean>> transformFipsPseudoRegionIfNecessary(Region region) {
```

Comment:
note: `Optional<Boolean>` because `Pair` doesn't allow values to be null

Reply:
Can we use False instead of null? Seems like currently we never have Optional(False).

Reply:
`false` seems weird and would wouldn't work just setting the value directly since now we'd be setting an explicit value for fips_enabled in cases where the region is "normal", and would supercede sourcing the FIPS value from the environment https://github.com/aws/aws-sdk-java-v2/blob/master/core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java#L324